source/BadgeOverlaySupport.js

```diff
@@ -0,0 +1,144 @@
+	*			{from: '.model.coverSource', to: '.source'}
+	*		]
+	*	}
+	* ]}
+	* ```
+	*
+	* The item may define a
+	* [badgeScrimIcon]{@link moon.BadgeOverlaySupport#badgeScrimIcon} URL to
+	* override the default icon.
+	*
+	* ```javascript
+	* {name: 'list', kind: 'moon.DataList', components: [
+	*	{mixins: ['moon.BadgeOverlaySupport'], type: 'video', badgeScrimIcon: 'assets/my-icon.png',
+	*		kind: 'moon.ImageItem', bindings: [
+	*			{from: '.model.title', to: '.label'},
+	*			{from: '.model.description', to: '.text'},
+	*			{from: '.model.coverSource', to: '.source'}
+	*		]
+	*	}
+	* ]}
+	* ```
+	*
+	* The item may define a
+	* [align]{@link moon.BadgeOverlaySupport#align} to
+	* specify position of badge. Possible values are 'top' or 'bottom'. Defaulted as 'bottom'.
+	* ```javascript
+	* {name: 'list', kind: 'moon.DataList', components: [
+	*	{mixins: ['moon.BadgeOverlaySupport'], type: 'video', align: 'top',
+	*		kind: 'moon.ImageItem', bindings: [
+	*			{from: '.model.title', to: '.label'},
+	*			{from: '.model.description', to: '.text'},
+	*			{from: '.model.coverSource', to: '.source'}
+	*		]
+	*	}
+	* ]}
+	*
+	* @mixin moon.BadgeOverlaySupport
+	* @protected
+	*/
+	moon.BadgeOverlaySupport = {
+
+		/**
+		* @private
+		*/
+		name: 'moon.BadgeOverlaySupport',
+
+		/**
+		* @private
+		*/
+		classes: 'moon-badge-overlay-support',
+
+		/**
+		* Identifies the MIME type of badge. 
+		* Possible values are "image", "audio" or "video".
+		* Defaulted as 'image'.
+		*
+		* @name moon.BadgeOverlaySupport#type
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+
+		/**
+		* URL for icon to be used in place of default icon.
+		*
+		* @name moon.BadgeOverlaySupport#badgeScrimIcon
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+
+		/**
+		* Sepcifies the alignment position of badge.
+		* Possible values are 'top' or 'bottom'.
+		* Defaulted as 'bottom'.
+		*
+		* @name moon.BadgeOverlaySupport#badgeScrimIcon
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		/**
+		* @method
+		* @private
+		*/
+		create: enyo.inherit(function (sup) {
+			return function () {
+				sup.apply(this, arguments);
+				if(this.type && this.type != "image") {
+					this.createChrome(this._badgeScrim);
+					// Allow the icon to be modified by user
+					if (this.badgeScrimIcon) {
+						this.$.badgeScrimIcon.set('icon','');
+					}
+					this.alignChanged();
+				}
+			};
+		}),
+
+		/**
+		* @private
+		*/
+		bindings: [
+			{from: '.badgeScrimIcon', to: '.$.badgeScrimIcon.src'}
+		],
+
+		/**
+		* @private
+		*/
+		_badgeScrim: [
+			{name:'badgeScrim', classes: 'enyo-fit moon-badge-overlay-support-scrim', components: [
+				{name:'badgeScrimIcon', kind: 'moon.Icon', small: false, icon: 'play', spotlight: false}
+			]}
+		],
+
+		alignChanged: function() {
+			if(this.$.badgeScrim) {
+				// TODO: identify the proper value based on height of image.
+				this.$.badgeScrim.applyStyle("top", this.align=='top'?0:'125px'); 
```

Comment:
We should avoid using hard-coded values in the JavaScript. This same effect could very easily be achieved by adding a class with the static rules written into the CSS files. It would also be preferable to have this not be a static value, something like `top: 50%` would work out better in the case of supporting multiple different sizes.